### PR TITLE
Various fixes based on Scrutinizer input.

### PR DESCRIPTION
**Pull request by jrfnl**
Ran the codebase through Scrutinizer and applied a lot of quick fixes based on the feedback. Mostly variable type checks and potential type juggling issues.

### CODE REVIEW

File: WordPress/AbstractArrayAssignmentRestrictionsSniff.php [hunk 1]
```diff
@@ -108,9 +108,11 @@ public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr ) {
 		if ( in_array( $token['code'], array( T_CLOSE_SQUARE_BRACKET, T_DOUBLE_ARROW ), true ) ) {
 			if ( T_CLOSE_SQUARE_BRACKET === $token['code']  ) {
 				$operator = $phpcsFile->findNext( array( T_EQUAL ), ( $stackPtr + 1 ) );
-			} elseif ( T_DOUBLE_ARROW === $token['code'] ) {
+			} else {
```
Comment (GaryJones) [inline WordPress/AbstractArrayAssignmentRestrictionsSniff.php]: This `else` could be the default:

```
if ( in_array( $token['code'], array( T_CLOSE_SQUARE_BRACKET, T_DOUBLE_ARROW ), true ) ) {
	// T_DOUBLE_ARROW.
	$operator = $stackPtr;
	if ( T_CLOSE_SQUARE_BRACKET === $token['code']  ) {
		$operator = $phpcsFile->findNext( array( T_EQUAL ), ( $stackPtr + 1 ) );
	}
```

File: WordPress/Sniff.php [hunk 2]
```diff
@@ -562,8 +562,8 @@ protected function has_whitelist_comment( $comment, $stackPtr ) {
 
 		$last = $this->tokens[ $lastPtr ];
 
-		if ( T_COMMENT === $last['code'] ) {
-			return preg_match( '#' . preg_quote( $comment ) . '#i', $last['content'] );
+		if ( T_COMMENT === $last['code'] && preg_match( '#' . preg_quote( $comment ) . '#i', $last['content'] ) === 1 ) {
```
Comment (GaryJones) [inline WordPress/Sniff.php]: Why not just straight return?

```
return T_COMMENT === $last['code'] && preg_match( '#' . preg_quote( $comment ) . '#i', $last['content'] ) === 1;
```

File: WordPress/AbstractArrayAssignmentRestrictionsSniff.php (hunk 1 above)
Comment (jrfnl) [inline WordPress/AbstractArrayAssignmentRestrictionsSniff.php]: Done

File: WordPress/Sniff.php (hunk 2 above)
Comment (jrfnl) [inline WordPress/Sniff.php]: Refactored to keep readability & still do a straight return.

**Comment by jrfnl** [comment]
Please ping me when someone is ready to merge this and I'll rebase asap - see https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/pull/637#issuecomment-242906170

**Pull request merged**